packages/markers/src/browser/marker-manager.ts

```diff
@@ -160,6 +172,22 @@ export abstract class MarkerManager<D extends object> {
         return oldMarkers;
     }
 
+    /*
+     * adds markers for the given uri and owner with the given data, without touching the exsting markers associated with the same uri and owner.
+     */
+    addMarkers(uri: URI, owner: string, data: D[]): Marker<D>[] {
```

Comment:
Could we rewrite the client instead that we have a single method `setMarker` which updates them and fire event instead of checking that logic is proper in different methods?

Reply:
`setMarkers()` removes existing markers before adding the new ones in.
`addMarkers()` adds new markers to the existing markers collection.
They are doing two different jobs.

I can definitely remove `addMarkers()`, and only use `setMarkers()`. The problem is that, if I have TWO running tasks that happen to have the same owner, markers from one task would overwrite makers from the other. And we will end up with having markers missing in the problem view.

Reply:
no, if you fetch markers for an owner first in taks extension and your markers and then set markers?

Reply:
ok i reverted changes made to the taskmanager api, and adjusted the client code. also added tests to the classes i added.